src/app/dim-ui/usePopper.ts

```diff
@@ -29,66 +33,69 @@ const createPopper = popperGenerator({
 });
 
 const popperOptions = (
-  placement: Options['placement'] = 'auto',
+  placement: Placement = 'auto',
+  fallbackPlacements: Placement[] = [],
   arrowClassName?: string,
-  boundarySelector?: string,
+  boundaryElement?: Element,
   offset = arrowClassName ? 5 : 0,
   fixed = false
 ): Partial<Options> => {
   const headerHeight = parseInt(
     document.querySelector('html')!.style.getPropertyValue('--header-height')!,
     10
   );
-  const boundaryElement = boundarySelector && document.querySelector(boundarySelector);
   const padding: Padding = {
     left: 10,
     top: headerHeight + (boundaryElement ? boundaryElement.clientHeight : 0) + 5,
     right: 10,
     bottom: 10,
   };
   const hasArrow = Boolean(arrowClassName);
-  return {
-    strategy: fixed ? 'fixed' : 'absolute',
-    placement,
-    modifiers: _.compact([
-      {
-        name: 'preventOverflow',
-        options: {
-          priority: ['bottom', 'top', 'right', 'left'],
-          boundariesElement: 'viewport',
-          padding,
-        },
-      },
-      {
-        name: 'flip',
-        options: {
-          behavior: ['top', 'bottom', 'right', 'left'],
-          boundariesElement: 'viewport',
-          padding,
-        },
-      },
-      {
-        name: 'offset',
-        options: {
-          offset: [0, offset],
-        },
-      },
-      hasArrow && {
+
+  const preventOverflowOptions: Partial<PreventOverflowModifier> = {
+    name: 'preventOverflow',
+    options: {
+      boundary: boundaryElement || 'clippingParents',
```

Comment:
Clipping parents probably isn't needed as its the default. Setting the viewport as the max boundary is done by the [rootBoundary](https://popper.js.org/docs/v2/modifiers/prevent-overflow/#rootboundary) which defaults to the document.